app/components/provider_interface/activity_log_event_component.rb

```diff
@@ -0,0 +1,46 @@
+module ProviderInterface
+  class ActivityLogEventComponent < ViewComponent::Base
+    include ViewHelper
+
+    attr_reader :event, :application_choice, :course_option
+
+    def initialize(activity_log_event:)
+      @event = activity_log_event
+      @application_choice = event.auditable
+      @course_option = @application_choice.offered_option
+    end
+
+    def event_description
+      user = event.user.try(:full_name) || event.user.try(:display_name)
```

Comment:
Maybe push this down onto the `event` itself? Slightly more cohesive

Reply:
I think safe navigation is a better option than a `try` block here.

```suggestion
      user = event.user&.full_name || event.user&.display_name
```

Reply:
`user` here is polymorphic (support users have `.display_name` and provider users have `.full_name`) -- using `&.` here would result in `NoMethodError`. I think `.try` is appropriate.